python/lsst/daf/base/baseLib.i

```diff
@@ -161,6 +161,48 @@ def _propertyContainerGet(container, name, asArray=False):
         pass
     raise lsst.pex.exceptions.TypeError('Unknown PropertySet value type for ' + name)
 
+import numbers
+def _guessIntegerType(container, name, value):
+    """Given an existing container and name, determine the type
+    that should be used for the supplied value. The supplied value
+    is assumed to be a scalar.
+
+    On Python 3 all ints are LongLong but we need to be able to store them
+    in Int containers if that is what is being used (testing for truncation).
+    Int is assumed to mean 32bit integer (2147483647 to -2147483648).
+
+    If there is no pre-existing value we have to decide what to do. For now
+    we pick Int if the value is less than maxsize.
+
+    Returns None if the value supplied is a bool or not an integral value.
+    """
+    useType = None
+    maxInt = 2147483647
+    minInt = -2147483648
+
+    # We do not want to convert bool to int so let the system work that
+    # out itself
+    if isinstance(value, bool):
+        return useType
+
+    if isinstance(value, numbers.Integral):
+        try:
+            containerType = _propertyContainerElementTypeName(container, name)
+        except lsst.pex.exceptions.NotFoundError:
+            # nothing in the container so choose based on size. Safe option is to
+            # always use LongLong
+            if value <= maxInt and value >= minInt:
+                useType = "Int"
+            else:
+                useType = "LongLong"
+        else:
+            if containerType == "Int":
+                if value <= maxInt and value >= minInt:
+                    useType = "Int"
+            elif containerType == "LongLong":
+                useType = "LongLong"
```

Comment:
Actually, there is a change of behavior on Python 3. On Python 2 "2147483649" is an `int` and not a `long` and so `addInt` is called which overflows. On Python 3 with the code above that falls through the type ends up as LongLong and when you call `addLongLong` on the `Int` item it triggers a type mismatch error. It seems that for compatibility I should check 64bit int range when the existing container type is Int and not the 32-bit range. This all seems to be a bit confused with Python `int` being a different size to PropertySet `Int`. I'll make the current behavior consistent with python 2 as it is indeed silly to shove a 64bit int into a 32-bit slot but I should let the later code tell you that.